sechub-sereco/src/main/java/com/daimler/sechub/sereco/importer/SarifV1JSONImporter.java

```diff
@@ -0,0 +1,181 @@
+
+                metaData.getVulnerabilities().add(vulnerability);
+            });
+        });
+
+        return metaData;
+    }
+
+    private SerecoCodeCallStackElement resolveCodeInfoFromResult(Result result) {
+
+        Optional<CodeFlow> codeFlows = result.getCodeFlows().stream().findFirst();
+
+        if (!codeFlows.isPresent()) {
+            // if no CodeFlow available, try to extract callstack directly from locations
+            return resolveCodeInfoFromLocations(result.getLocations());
+        }
+
+        Optional<ThreadFlow> optFlow = codeFlows.get().getThreadFlows().stream().findFirst();
+
+        if (!optFlow.isPresent()) {
+            return null;
+        }
+
+        ThreadFlow flow = optFlow.get();
+
+        List<Location> locations = flow.getLocations().stream().map(location -> location.getLocation()).collect(Collectors.toList());
+
+        return resolveCodeInfoFromLocations(locations);
+    }
+
+    private SerecoCodeCallStackElement resolveCodeInfoFromLocations(List<Location> locations) {
+
+        List<SerecoCodeCallStackElement> callstack = callStackListFromLocations(locations);
+
+        SerecoCodeCallStackElement firstElement = null;
+        SerecoCodeCallStackElement prevElement = null;
+
+        for (SerecoCodeCallStackElement code : callstack) {
+            if (firstElement == null) {
+                firstElement = code;
+            } else if (prevElement == null) {
+                firstElement.setCalls(code);
+                prevElement = code;
+            } else {
+                prevElement.setCalls(code);
+                prevElement = code;
+            }
+        }
+
+        return firstElement;
+    }
+
+    private List<SerecoCodeCallStackElement> callStackListFromLocations(List<Location> locations) {
+
+        List<SerecoCodeCallStackElement> callstack = new ArrayList<>();
+
+        if (locations == null) {
+            return callstack;
+        }
+
+        locations.forEach(location -> {
+
+            PhysicalLocation physicalLocation = location.getPhysicalLocation();
+
+            SerecoCodeCallStackElement subCode = new SerecoCodeCallStackElement();
+            subCode.setLocation(physicalLocation.getArtifactLocation().getUri());
+            subCode.setLine(physicalLocation.getRegion().getStartLine());
+            subCode.setColumn(physicalLocation.getRegion().getStartColumn());
+
+            callstack.add(subCode);
+        });
+        return callstack;
+    }
+
+    private String cweIdFrom(Result result) {
+
+        Optional<ReportingDescriptorReference> taxa = result.getTaxa().stream().filter(comp -> comp.getToolComponent().getName() == "CWE").findFirst();
+
+        if (!taxa.isPresent()) {
+            return "";
+        }
+
+        return taxa.get().getId();
+    }
+    
+    private List<Result> resultsWithGroupedLocations(List<Result> results) {
+        
+        HashMap<Integer, Result> mappedResults = new HashMap<>();
```

Comment:
Please use the generic `Map` instead of `HashMap` for the `mappedResults`.

Reply:
done